BizHawk.Client.Common/lua/EmuLuaLibrary.Events.cs

```diff
@@ -183,9 +183,9 @@ public string OnLoadState(LuaFunction luaf, string name = null)
 			return nlf.Guid.ToString();
 		}
 
-		[LuaMethodExample("local steveonm = event.onmemoryexecute(\r\n\tfunction()\r\n\t\tconsole.log( \"Fires after the given address is executed by the core\" );\r\n\tend\r\n\t, 0x200, \"Frame name\", \"System Bus\" );")]
-		[LuaMethod("onmemoryexecute", "Fires after the given address is executed by the core")]
-		public string OnMemoryExecute(LuaFunction luaf, uint address, string name = null, string domain = null)
+		[LuaMethodExample("local steveonm = event.onmemoryexecute(\r\n\tfunction()\r\n\t\tconsole.log( \"Fires after the given address is executed by the core. If no address is given, it will attach to every memory read\" );\r\n\tend\r\n\t, 0x200, \"Frame name\", \"System Bus\" );")]
+		[LuaMethod("onmemoryexecute", "Fires after the given address is executed by the core. If no address is given, it will attach to every memory read")]
+		public string OnMemoryExecute(LuaFunction luaf, uint? address = null, string name = null, string domain = null)

```

Comment:
The reason we throw an error originally is my concern is that it would be easy to leave out an address in a script, and that it would have huge concequences (a callback on all executes is insanely slow, depending on core).

I think address should not have a default value here and make the user explicitly add nil if they want this functionality.

Other than this small change, I think this is a great PR :)

Reply:
Thanks!

Would it be sufficient to just remove the default value from the header? Because in Lua, explicit nil is indistinguishable from a missing parameter. I'm not super familiar with how the bridging works.

Maybe in the future we could have explicit registerall and even registerrange type callbacks to prevent this 100%.

Reply:
We're not sure either. Please test both cases.

Reply:
Can you confirm that omitting the second parameter terminates the script rather than filling it with `null`? Might want to try both Lua engines.

Reply:
Haven't had the chance to test this, sorry. Hopefully I can get to it tonight